DSCResources/MSFT_xNetBIOS/MSFT_xNetBIOS.psm1

```diff
@@ -0,0 +1,137 @@
+try
+{
+    [void][reflection.assembly]::GetAssembly([NetBIOSSetting])
+}
+catch
+{
+    Add-Type -TypeDefinition @'
+    public enum NetBiosSetting
+    {
+       Default,
+       Enable,
+       Disable
+    }
+'@
+}
+
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $InterfaceAlias,
+
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Default","Enable","Disable")]
+        [System.String]
+        $Setting
+    )
+    try
+    {
+        $Netadapterparams = @{
+            ClassName = 'Win32_NetworkAdapter'
+            Filter = 'NetConnectionID="{0}"' -f $InterfaceAlias
+        }
+        $NetAdapterConfig = Get-CimInstance @Netadapterparams | 

```

Comment:
For consistency with the other resources it would be good to remove the try/catch and set the -ErrorAction Stop parameter on the Get-Cim* cmdlets. E.g.
```powershell
$Netadapterparams = @{
    ClassName = 'Win32_NetworkAdapter'
    Filter = 'NetConnectionID="{0}"' -f $InterfaceAlias
}
$NetAdapterConfig = Get-CimInstance `
     @Netadapterparams `
     -ErrorAction Stop | 
     Get-CimAssociatedInstance `
            -ResultClassName Win32_NetworkAdapterConfiguration `
            -ErrorAction Stop
```